src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/FormContext.cs

```diff
@@ -52,6 +51,19 @@ public IList<IHtmlContent> EndOfFormContent
 
         public bool CanRenderAtEndOfForm { get; set; }
 
+        private Dictionary<string, bool> RenderedFields
+        {
+            get
+            {
+                if (_renderedFields == null)
+                {
+                    _renderedFields = new Dictionary<string, bool>(StringComparer.Ordinal);
+                }
+
+                return _renderedFields;
+            }
+        }
```

Comment:
This makes the code _look_ more performant but I doubt it matters beyond the dictionary for the outer ("non-form") `FormContext`. Most forms contain generated fields :smiling_imp: Unless we're optimizing at the level of one `Dictionary<,>` per request, I suggest leaving `_renderedFields` alone since it was simpler

On the other hand, the removal of a `new FormContext()` when calling the `ViewContext` copy constructor should be a significant win.